std/assembly/string.ts

```diff
@@ -69,6 +69,7 @@ import { idof } from "./builtins";
   }
 
   concat(other: String): String {
+    assert(this !== null);
```

Comment:
Currently trying to get away from these. There are still scenarios where this can happen, like uninitialized class fields, but with initial null checking in place a lot is caught already.

Reply:
Removed